pkg/controller/controller.go

```diff
@@ -605,3 +609,46 @@ func unsetLoggerCtxValue(ctx context.Context, key string) context.Context {
 
 	return ctx
 }
+
+func reportLastReconciled(o interface{}) {
+	var kind string
+	{
+		obj, ok := o.(pkgruntime.Object)
+		if !ok {
+			return
```

Comment:
I don't think it's worth printing errors or return an error for metrics since it's not a major business of the controller. WDYT?

Reply:
What happens if the metrics are not emitted? Will the alerts based on the last reconciled timestamp still fire?

If so then I think thats OK. As long as it's clear how to debug this.

Reply:
>  Will the alerts based on the last reconciled timestamp still fire?

Yes, exactly. Alerts would be based on the last reconciled timestamp.